lib/sha256.c

```diff
@@ -192,7 +192,7 @@ static void SHA256_Transform(uint32_t *state, const unsigned char block[64])
 	/* Clean the stack. */
 	memset(W, 0, 256);
 	memset(S, 0, 32);
-	t0 = t1 = 0;
+	t0 = t1 = 0; t0++; t1++;
```

Comment:
these need to go on newlines per linux kernel style

Reply:
actually on second reading why not
`t0 = t1 = 1`?

Reply:
Nope, clang needs to see t0 and t1 to be used to stop warning.

Reply:
I agree w/ Vincent here.  This is the right thing to do.  It's unfortunate that the clang SA has no way to tell it we are doing this on purpose for security reasons.

Reply:
Alright, makes sense. I suggest we use the following instead of setting the values to 1, as it lines up with the preceding code and keeps the contents of the stack consistent. Colin probably had a good reason to set things to zero.

```
        memset(t0, 0, 4);
        memset(t1, 0, 4);
```

Reply:
Sometimes the void casting trick works to avoid the warning:
```
t0 = t1 = 0; (void)t0; (void)t1;
```

Reply:
If you choose the `memset` one, I would at least put a comment before, as it's kind of ugly to read.